### PR TITLE
[Merged by Bors] - feat: define the category of extremally disconnected compact Hausdorff spaces

### PR DESCRIPTION
**Pull request by dagurtomas**
Co-authored-by: Adam Topaz [topaz@ualberta.ca](mailto:topaz@ualberta.ca) @adamtopaz 
Basics of the category `Stonean` of extremally disconnected compact Hausdorff spaces. This work was done during the 2023 Copenhagen masterclass on formalisation of condensed mathematics. Numerous participants contributed.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Category/ExtrDisc/ExplicitLimits.lean
Comment (adamtopaz) [inline Mathlib/Topology/Category/ExtrDisc/ExplicitLimits.lean:148]: All this stuff can be shortened using the `cocone` and `isColimit` constructions above.

File: Mathlib/Topology/Category/ExtrDisc/ExplicitLimits.lean
Comment (adamtopaz) [inline Mathlib/Topology/Category/ExtrDisc/ExplicitLimits.lean:36]: This should go in the `ExtremallyDisconnected` file.

File: Mathlib/Topology/Category/ExtrDisc/Basic.lean
Comment (adamtopaz) [inline Mathlib/Topology/Category/ExtrDisc/Basic.lean:154]: Since `inferInstance` works, this is probably not necessary.

File: Mathlib/Topology/Category/ExtrDisc/Basic.lean
Comment (adamtopaz) [inline Mathlib/Topology/Category/ExtrDisc/Basic.lean:136]: We should add a suitable instance for topological spaces, and use that instance here.

File: Mathlib/Topology/Category/ExtrDisc/Basic.lean
Comment (adamtopaz) [inline Mathlib/Topology/Category/ExtrDisc/Basic.lean:53]:
```suggestion
/-- Projective implies ExtrDisc. -/
```

File: Mathlib/Topology/Category/ExtrDisc/ExplicitLimits.lean
Comment (adamtopaz) [inline Mathlib/Topology/Category/ExtrDisc/ExplicitLimits.lean:79]: I don't think `surj` is needed, is it?

**Comment by dagurtomas** [comment]
I've modified everything according to Adam's suggestions

File: Mathlib/Topology/Category/ExtrDisc/Basic.lean
Comment (semorrison) [inline Mathlib/Topology/Category/ExtrDisc/Basic.lean:49]: Hmm... I'd say they do. :-)

File: Mathlib/Topology/Category/ExtrDisc/Basic.lean
Comment (semorrison) [inline Mathlib/Topology/Category/ExtrDisc/Basic.lean:53]:
```suggestion
/-- `Projective` implies `ExtrDisc`. -/
```

File: Mathlib/Topology/Category/ExtrDisc/Basic.lean
Comment (semorrison) [inline Mathlib/Topology/Category/ExtrDisc/Basic.lean:138]:
```suggestion
  obj X := 
  { toCompHaus := X.compHaus,
```
Very minor...

File: Mathlib/Topology/ExtremallyDisconnected.lean
Comment (semorrison) [inline Mathlib/Topology/ExtremallyDisconnected.lean:154]: These should live somewhere earlier. Try `#find_home sigma_mk_preimage_image'`.

File: Mathlib/Topology/Category/ExtrDisc/Basic.lean
Comment (semorrison) [inline Mathlib/Topology/Category/ExtrDisc/Basic.lean:102]:
```suggestion
```

File: Mathlib/Topology/Category/ExtrDisc/Basic.lean
Comment (semorrison) [inline Mathlib/Topology/Category/ExtrDisc/Basic.lean:110]:
```suggestion
instance : Faithful toCompHaus := {}
```

File: Mathlib/Topology/Category/ExtrDisc/Basic.lean
Comment (semorrison) [inline Mathlib/Topology/Category/ExtrDisc/Basic.lean:121]: I don't object to this style at all, but some people prefer to write `inferInstanceAs <| TopologicalSpace X.compHaus`.

File: Mathlib/Topology/Category/ExtrDisc/Basic.lean
Comment (semorrison) [inline Mathlib/Topology/Category/ExtrDisc/Basic.lean:153]: This shouldn't be an instance!!

File: Mathlib/Topology/Category/ExtrDisc/Basic.lean
Comment (semorrison) [inline Mathlib/Topology/Category/ExtrDisc/Basic.lean:144]:
```suggestion
instance : Full toProfinite where
  preimage f := f
```
more readable!

File: Mathlib/Topology/Category/ExtrDisc/Basic.lean
Comment (semorrison) [inline Mathlib/Topology/Category/ExtrDisc/Basic.lean:148]:
```suggestion
instance : Faithful toProfinite := {}
```

File: Mathlib/Topology/Category/ExtrDisc/ExplicitLimits.lean
Comment (semorrison) [inline Mathlib/Topology/Category/ExtrDisc/ExplicitLimits.lean:1]: Same comment about the file name.

**Comment by jcommelin** [comment]
@dagurtomas Please click "resolve conversation" on all the conversations above that you consider done. (Unless you still want a maintainer to look/reply to a conversation.)

**Comment by dagurtomas** [comment]
Merging master and waiting for CI to build to be able to use `#find_home`

File: Mathlib/Topology/Category/Stonean/Basic.lean
Comment (jcommelin) [inline Mathlib/Topology/Category/Stonean/Basic.lean:28]: This mostly repeats line 17. I think you can remove this paragraph (and maybe add the "basic observations about ..." bit to the paragraph on L17.

File: Mathlib/Topology/Category/Stonean/Basic.lean
Comment (jcommelin) [inline Mathlib/Topology/Category/Stonean/Basic.lean:75]: I think this should be factored out into a separate instance. In Lean 4 it is not a problem to have cycles in the type class hierarchy. So for `X : CompHaus` I think we should have a cycle `extrDisc -> Projective -> extrDisc`.

**Comment by jcommelin** [comment]
Two comments, but apart from that, LGTM

**Comment by dagurtomas** [comment]
> Two comments, but apart from that, LGTM

Done

**Comment by dagurtomas** [comment]
Hmm, CI canceled the check "Detect changes to header SHAs / Add annotations (pull_request)". What does that mean?

**Comment by RemyDegenne** [comment]
That check broke following a change in the way we do those headers. Please merge master. That should fix it.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**
